torch/_inductor/codegen/wrapper.py

```diff
@@ -505,7 +510,18 @@ def add_benchmark_harness(self, output):
                 )
             output.writeline("else:")
             with output.indent():
-                output.writeline("benchmark_compiled_module()")
+                output.writeline("with maybe_profile(args.profile) as p:")
+                with output.indent():
+                    output.writeline("benchmark_compiled_module()")
+                output.writeline("")
+                output.writeline("if p:")
+                with output.indent():
+                    output.writelines([
+                        'fd, path = tempfile.mkstemp(prefix="compiled_module_profile_", suffix=".json")',
```

Comment:
I'm not sure making tempfile everytime is needed, you can output to some fixed filename and overwrite on rerun.